adapters/tlsio_openssl.c

```diff
@@ -1177,7 +1349,64 @@ static X509_CRL *load_crl_crldp(X509 *cert, const char* suffix, STACK_OF(DIST_PO
             continue;
         }
 
-        // at this point, we got a valid crl
+        *pCrl = crl;
+        ret = 1;
+    }
+
+    return ret;
+}
+
+static int save_cert_crl_file(X509 *cert, const char* suffix, X509_CRL *crl)
```

Comment:
in many functions you return error code, but these are ignored on the calling side in several places. i'm not sure why you need then to return anything...